src/hypertable.h

```diff
@@ -17,13 +18,22 @@ typedef struct Hypertable
 	Oid			main_table_relid;
 	Hyperspace *space;
 	SubspaceStore *chunk_cache;
+	int			num_tablespaces;
+	Tablespace *tablespaces;
```

Comment:
Why are we adding Tablespace to the hypertable and, by extension, the hypertable cache. Tablespaces are not used on any of the fast paths as far as I can tell. Adding it to the cache increases complexity and I don't see why it's needed.

Reply:
I guess I see things differently from you in this regard. To me, this simplifies a lot of code, including tablespace lookups for attach/detach. It also unifies how things work, avoiding the confusing mix of C/PLPGSQL and upcalls/downcalls . Further, it makes tablespace selection much simpler (upcoming PR).

It also falls out naturally from moving to a unified catalog API, where catalog updates happen in one consistent way.

Reply:
Why is it harder to just have a function to fetch the Tablespace collection given a hypertable and not include in the cache? I don't see why C/PLPGSQL and upcalls/downcalls has anything to do with this? Nor do I see how this connects to a unified catalog API.